lib/evaluate/fold.cc

```diff
@@ -966,8 +966,8 @@ static std::optional<Constant<SubscriptInteger>> GetConstantSubscript(
             }
           },
           [](Triplet &triplet) -> std::optional<Constant<SubscriptInteger>> {
-            std::optional<std::int64_t> lbi{1}, ubi;
-            std::optional<std::int64_t> stride{ToInt64(triplet.stride())};
+            std::optional<ConstantSubscript> lbi{1}, ubi;
+            std::optional<ConstantSubscript> stride{ToInt64(triplet.stride())};
```

Comment:
Using `ToInt64` here seems to break the abstraction. If `ConstantSubscript` were a wider type than `std::int64_t` we wouldn't get all of the bits.

Ideally we would want `std::optional<ConstantSubscript> ToConstantSubscript(...)`, as well as `ToLengthCIntType()` and `ToSubscriptCIntType()`, but that seems excessive.

Reply:
Tell me what you want.

Reply:
What do you think of changing `ToInt64` to `std::optional<std::intmax_t> ToIntMax(...)`? Then we know whatever `SubscriptCIntType` is, we won't lose precision in this kind of initialization.

If you don't think that is worth the trouble, this is fine as it is.

Reply:
Braced initialization would become errors as those 128-bit integer values were used to initialize `ConstantSubscript` values that were 64 bits.  So I'd have to put casts in, or an intervening conversion function call, or change `ConstantSubscript` to be 128 bits.